### PR TITLE
[Logs]: Ensure the logs-scheduler is not blocking

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

* Don't send sources and services to channels if there are no consumers instead of always consuming.
* Cleaned launchers.
* Added unit-tests

### Motivation

Today, the logs-scheduler can be blocking and freeze the autodiscovery if a new config is scheduled and no component consumes it.

### Additional Notes

Agent 6.5 Post Mortem follow up

### CODE REVIEW

File: pkg/logs/config/config.go [hunk 1]
```diff
@@ -29,6 +30,17 @@ func DefaultSources() []*LogSource {
 		})
 		sources = append(sources, source)
 	}
+
+	if LogsAgent.GetBool("logs_config.container_collect_all") {
+		// append a new source to collect all logs from all containers
+		source := NewLogSource("container_collect_all", &LogsConfig{
```
Comment (achntrl) [inline pkg/logs/config/config.go]: Why do we add a Docker source ? When starting collecting logs on k8 running on containerd, we will have something like this 

```
kube-system/kube-controller-manager/kube-controller-manager
  -----------------------------------------------------------
    Type: file
    Path: /var/log/pods/8d989b65-ba83-11e8-b67f-025f4577d1f5/kube-controller-manager/*.log
    Status: OK
    Inputs: /var/log/pods/8d989b65-ba83-11e8-b67f-025f4577d1f5/kube-controller-manager/0.log

  container_collect_all
  ---------------------
    Type: docker
    Status: Pending
```

Comment (ajacquemot) [reply pkg/logs/config/config.go]: Agreed, it's a corner case, I don't have a good solution to address this for now. Any idea ?

Comment (achntrl) [reply pkg/logs/config/config.go]: Not for now

**Review by achntrl — APPROVED** [review]
👍

**Pull request merged**